dart/lib/flat_buffers.dart

```diff
@@ -41,6 +41,14 @@ class BufferContext {
   @pragma('vm:prefer-inline')
   Uint8List _asUint8List(int offset, int length) =>
       _buffer.buffer.asUint8List(_buffer.offsetInBytes + offset, length);
+  
+  @pragma('vm:prefer-inline')
+  Int8List _asInt8List(int offset, int length) =>
+      _buffer.buffer.asInt8List(_buffer.offsetInBytes + offset, length);
+
+  @pragma('vm:prefer-inline')
+  Uint16List _asUint16List(int offset, int length) =>
+      _buffer.buffer.asUint16List(_buffer.offsetInBytes + offset, length);
```

Comment:
As with the other comment below, this ony has a valid result if running on little-endian platforms.
```suggestion
  Uint16List _asUint16List(int offset, int length) {
      assert(Endian.host == Endian.little);
      _buffer.buffer.asUint16List(_buffer.offsetInBytes + offset, length);
  }
```